content/using_ec2_spot_instances_with_eks/deploy/tolerations_and_affinity.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Tolerations and Affinity"
+date: 2018-08-07T08:30:11-07:00
+weight: 20
+---
+
+## Adding Tolerations
+
+Our next task is to add tolerations and affinities to the application.
+
+In the previous section we created two nodegroups that were tainted with 
+`spotInstance: "true:PreferNoSchedule"`. **PreferNoSchedule** is used to indicate we prefer pods not be scheduled on Spot Instances. **NoSchedule** can also be used to enforce a hard discrimination as a taint.   To overcome the `spotInstance: "true:PreferNoSchedule"` taint, we need to create a toleration in the deployment. Read about how [tolerations](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/) are applied and modify the **monte-carlo-pi-service.yml** file accordingly.
+
+
+{{%expand "Show me a hint for implementing this." %}}
+As per the [tolerations](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/) documentation 
+the objective is to add the following section to the `monte-carlo-pi-service.yml`. 
+
+The following toleration must be added at the *spec.template.spec* level
+
+```
```

Comment:
I'd say providing more then the little sniplet to add will make it more readable.
```
  spec:
      tolerations:
      - key: "spotInstance"
        operator: "Equal"
        value: "true"
        effect: "PreferNoSchedule"
      containers:
        - name: monte-carlo-pi-service
```

Reply:
Same reason as below. I think It might be misleading on which parts they copy paste. Specially given we make copy paste so easy. Users have the solution file as a reference.